### ISSUE TITLE
Bug: [new config system] Unexpected glob behavior in `FlatESLint`

### ISSUE THREAD

**Issue by fisker** [eslint/eslint#16340]
### Environment

Node version: 18.8.0
npm version:  N/A
Local ESLint version: 8.23.1
Global ESLint version: A/A
Operating System: Windows

### What parser are you using?

Default (Espree)

### What did you do?

Run `FlatESLint` with different cwd on same file.

### What did you expect to happen?

Should result the same.

### What actually happened?

They result differently. 

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

Step to reproduce:
1. Clone the repo https://github.com/fisker/flat-eslint-cmd-bug-repo (or run on [gitpod.io](https://gitpod.io/#https://github.com/fisker/flat-eslint-cmd-bug-repo))
1. run `yarn && node test.mjs` to see the difference

It seems [the `globby` option `{absolute: true}`](https://github.com/eslint/eslint/blob/2c152ff0fb709b99e62c19ecd2c95689efacbe4c/lib/eslint/eslint-helpers.js#L231) cause `globby` can't match any file.

Remove that line cann't really fix my problem, but it should be another issue, possible related to #16264

Since the globby version we are using is really old, not sure what causes that. (I'm a maintainer of `globby`)

**Comment by mdjermanovic** [comment]
Only `test/integration/fixtures/vscode` cwd loads `test/integration/fixtures/vscode/.eslintignore`. That works as intended.

However, the patterns in that file should not ignore the `src/bootstrap-amd.js` file, but they do. In particular, it's ignored by [`**/fixtures/**`](https://github.com/fisker/flat-eslint-cmd-bug-repo/blob/b5d83269d9e6afe90d28169e8f0ac9acf4caff09/test/integration/fixtures/vscode/.eslintignore#L17) pattern.

This is either a bug or a surprising behavior in `globby`/`fast-glob`. We expect that the patterns in `ignore` will be interpreted as relative to `cwd`, but `{ cwd: "/path/to/test/integration/fixtures/vscode", ignore: ["**/fixtures/**"], absolute: true }` effectively ignores everything, probably because the ignore pattern matches absolute paths.

**Comment by mdjermanovic** [comment]
> Since the globby version we are using is really old, not sure what causes that. (I'm a maintainer of `globby`)

I'm getting the same behavior with the latest globby v13.1.2:

```js
// globby-test-ignore/test.mjs

import { globby } from 'globby';

const paths = await globby(["**/*"], {
    ignore: ["**/globby-test-ignore/**"],
    absolute: true
});

console.log(paths); // []
```

Is this the intended behavior?

**Comment by fisker** [comment]
> it's ignored by **/fixtures/** pattern.

No wonder I can't reproduce when trying to make the reproduction smaller. (Took me a while to setup)

**Comment by mdjermanovic** [comment]
> > Is this the intended behavior?
> 
> Are you running the script inside `globby-test-ignore` or outside?

Inside, `cwd` ends with `/globby-test-ignore`.

I would expect that `**/globby-test-ignore/**` ignores `/my/absolute/path/to/globby-test-ignore/**/globby-test-ignore/**`, but it effectively ignores `/my/absolute/path/to/globby-test-ignore/**`

**Comment by fisker** [comment]
Sounds like a bug to me.  But seems absolute ignore is expected? https://github.com/mrmlnc/fast-glob/issues/113

**Comment by mdjermanovic** [comment]
I can reproduce this behavior with `ignores` in our [`eslint.config.js`](https://github.com/eslint/eslint/blob/main/eslint.config.js).

```diff
-    }
+    },
+    {
+        ignores: ["**/eslint/**"]
+    }
```

(my local clone is in a directory named `eslint`).

```
npx eslint *.js
```

```
Oops! Something went wrong! :(

ESLint: 8.23.1

You are linting "*.js", but all of the files matching the glob pattern "*.js" are ignored.

If you don't want to lint these files, remove the pattern "*.js" from the list of arguments passed to ESLint.

If you do want to lint these files, try the following solutions:

* Check your .eslintignore file, or the eslintIgnore property in package.json, to ensure that the files are not configured to be ignored.
* Explicitly list the files from this glob that you'd like to lint on the command-line, rather than providing a glob as an argument.
```

**Comment by nzakas** [comment]
Can someone summarize what the problem is here? I've re-read this a couple of times and can't quite tell how to create a simple repro case.

**Comment by mdjermanovic** [comment]
Here's a simplified example:

```
foo/
└── eslint.config.js
```

```js
// eslint.config.js
module.exports = [{
    ignores: ["**/foo/**"]
}];
```

Running `eslint "*.js"` in the directory `foo`. This is project's root dir,  where the `eslint.config.js` file is. `basePath` = `cwd` = `"/path/to/foo"`.

Path to the project dir and project dir's name shouldn't matter, so the expected result is to lint `eslint.config.js` file.

Actual result:

```
Oops! Something went wrong! :(

ESLint: 8.24.0

You are linting "*.js", but all of the files matching the glob pattern "*.js" are ignored.

If you don't want to lint these files, remove the pattern "*.js" from the list of arguments passed to ESLint.

If you do want to lint these files, try the following solutions:

* Check your .eslintignore file, or the eslintIgnore property in package.json, to ensure that the files are not configured to be ignored.
* Explicitly list the files from this glob that you'd like to lint on the command-line, rather than providing a glob as an argument.
```

because ignore pattern `**/foo/**`  matches absolute path `/path/to/foo/eslint.config.js`.

**Comment by nzakas** [comment]
Got it, thank you! Working on this now
